### ISSUE TITLE
get to v1.0.0

### ISSUE THREAD

**Issue by ghoneycutt** [voxpupuli/metadata-json-lint#35]
Since this is a dependency of puppetlabs_spec_helper and is widely used, a roadmap to get to v1.0.0 should be planned.

**Comment by nibalizer** [comment]
I think the answer to that is: release 1.0.0

**Comment by ghoneycutt** [comment]
Does anyone have any breaking changes that they want to get in? Which ruby versions will be supported?

**Comment by nibalizer** [comment]
https://github.com/voxpupuli/metadata-json-lint/pull/37

**Comment by nibalizer** [comment]
I do think we should be clear about what a breaking change for us is though. If the metadata.json spec itself changes, and m-j-l changes to reflect that, is that a version bump for m-j-l? I dont think so but I wonder if anyone disagrees.

**Comment by rnelson0** [comment]
#38 has some things #37 doesn't, and vice versa. I've reviewed both with my comments.

As for metadata.json specs, I think we wait and see if it ever happens. But I suspect it would coincide with a major version change in puppet, and then it might make sense to do a major version bump ourselves so we could say "m-j-l < 2.0 pairs with puppet < 5.0; m-j-l >= 2.0 pairs with puppet >= 5.0" or similar.

**Comment by ghoneycutt** [comment]
I based my work off of PR #37 so @nibalizer could incorporate it into the next release. If other people have changes, they could also base off of that branch.

**Comment by ghoneycutt** [comment]
Regarding versioning, both of your approaches seems fine with me. Perhaps you could take this to the mailing list and see what the rest of the community thinks.

**Comment by nibalizer** [comment]
So far this is what we've got for metadata.json spec. https://docs.puppet.com/guides/style_guide.html#module-metadata

**Comment by rnelson0** [comment]
Is anyone aware of breaking changes to the format, or was that just a theoretical? If not, we should split that into a new issue.

**Comment by ghoneycutt** [comment]
I was asking if anyone wants to make breaking changes to the m-j-l code base before it goes 1.0.0.

**Comment by rnelson0** [comment]
Looking at open issues, #18 could be added without a breaking-change. #26 I think would be - but I am inclined to close it and proceed with author being a required field both per documentation and, like license, as requiring a deliberate value instead of implicit value. Do either of you see a reason to keep #26 open?

If not, then #37/#38 look good to me, we just need to merge 38 first.

**Comment by ghoneycutt** [comment]
WRT #26 I agree that it should be removed from the required list, as it is not required as per the docs. That should happen before v1.

**Comment by rnelson0** [comment]
We're live! https://rubygems.org/gems/metadata-json-lint/ The publish date is wrong and I've opened a PR to fix that in the future. I'm not aware of any way to fix that on rubygems short of yank and publish 1.0.1 but I don't think it's that vital myself. If it is, let me know and I'll bump version and republish before closing this.
